src/problem.jl

```diff
@@ -1,13 +1,63 @@
 abstract type AbstractOptimizationProblem end
 
-struct OptimizationProblem{F,X,P,B,K} <: AbstractOptimizationProblem
+struct OptimizationProblem{F,X,P,B,LC,UC,K} <: AbstractOptimizationProblem
     f::F
     x::X
     p::P
     lb::B
     ub::B
+    lcons::LC
+    ucons::UC
     kwargs::K
-    function OptimizationProblem(f, x; p=DiffEqBase.NullParameters(), lb = nothing, ub = nothing, kwargs...)
-        new{typeof(f), typeof(x), typeof(p), typeof(lb), typeof(kwargs)}(f, x, p, lb, ub, kwargs)
+    function OptimizationProblem(f, x; p=DiffEqBase.NullParameters(), lb = [], ub = [], lcons = [], ucons = [], kwargs...)
+        new{typeof(f), typeof(x), typeof(p), typeof(lb), typeof(lcons), typeof(ucons), typeof(kwargs)}(f, x, p, lb, ub, lcons, ucons, kwargs)
     end
 end
+
+import MathOptInterface
+const MOI = MathOptInterface
+
+struct MOIOptimizationProblem <: MOI.AbstractNLPEvaluator
+    prob::OptimizationProblem
+end
+
+MOI.eval_objective(moiproblem::MOIOptimizationProblem, x) = moiproblem.prob.f isa OptimizationFunction ? moiproblem.prob.f.f(x) : moiproblem.prob.f(x)
```

Comment:
Even if the optimiser was evaluating the objective at the initial point `[0.0,0.0]` the objective value  should be one here. `0.0` looks like a default value being used.